### PR TITLE
[SPARK-32860][DOCS][SQL] Updating documentation about map support in Encoders

### PR DESCRIPTION
**Pull request by hannahkamundson**
### What changes were proposed in this pull request?

Javadocs updated for the encoder to include maps as a collection type

### Why are the changes needed?

The javadocs were not updated with fix SPARK-16706

### Does this PR introduce _any_ user-facing change?

Yes, the javadocs are updated

### How was this patch tested?

sbt was run to ensure it meets scalastyle

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
ok to test

File: sql/catalyst/src/main/scala/org/apache/spark/sql/Encoders.scala [hunk 1]
```diff
@@ -146,7 +146,8 @@ object Encoders {
    *  - String
    *  - java.math.BigDecimal, java.math.BigInteger
    *  - time related: java.sql.Date, java.sql.Timestamp, java.time.LocalDate, java.time.Instant
-   *  - collection types: only array and java.util.List currently, map support is in progress
+   *  - collection types: array, java.util.List, and map
+   *  - map is only available since version 2.1.0
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/Encoders.scala:150]: I think you can just remove this line. Spark versions<2.4 is EOL anyway

Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/Encoders.scala:150]:
```suggestion
```

**Review by HyukjinKwon — APPROVED** [review]

**Review by viirya — APPROVED** [review]
Looks good after @HyukjinKwon's suggested patch.

**Comment by HyukjinKwon** [comment]
Merged to master, branch-3.0 and branch-2.4.

**Pull request closed without merging**